### PR TITLE
Feature/us11621 progress bar

### PR DESCRIPTION
**Pull request by JoeKer1**
image

### CODE REVIEW

File: src/app/components/question-form/question-form.component.spec.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ describe("QuestionFormComponent", () => {
       );
       mockAssessmentService = jasmine.createSpyObj<AssessmentService>(
         "assessmentService",
-        ["SaveResponse", "GetRequiredResponses"]
+        ["SaveResponse", "GetRequiredResponses", "GetTotalQuestions"]
```
Comment (Doug-Shannon) [inline src/app/components/question-form/question-form.component.spec.ts]: questions?  thought these were responses

File: src/app/components/question-form/question-form.component.ts [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
-  private responses: WfResponse[];
   public answers: Answer[];
-  public ready: boolean = false;
+  public response: WfResponse;
+  public totalQuestions: number;
+  public totalQuestionsAnswered = null;
+  public ready = false;
   private assessmentDocumentID = null;
+  private responses: WfResponse[];
+  private mod: number;
```
Comment (Doug-Shannon) [inline src/app/components/question-form/question-form.component.ts]: full name for clarity?

File: src/app/components/question-progress-bar/question-progress-bar.component.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  public GetPercentage() {
+    if (this.answered === 0) {
+      return 0 + "%";
+    } else {
+      return this.answered / this.max * 100 + "%";
+    }
+  }
+
+  public getRandomEncouragement(lastShownTitle = "") {
+    const content = this.encouragements[
```
Comment (Doug-Shannon) [inline src/app/components/question-progress-bar/question-progress-bar.component.ts]: what happens if they only have one encouragement in the database?

File: src/app/services/assessment.service.ts [hunk 4]
```diff
@@ -65,6 +65,10 @@ export class AssessmentService {
     }
   }
 
+  public GetTotalQuestions(): number {
```
Comment (Doug-Shannon) [inline src/app/services/assessment.service.ts]: what if there is no assessment stored?

File: src/app/components/question-form/question-form.component.spec.ts (hunk 1 above)
Comment (JoeKer1) [reply src/app/components/question-form/question-form.component.spec.ts]: every component name refers to questions so ¯\_(ツ)_/¯

File: src/app/services/assessment.service.ts (hunk 4 above)
Comment (JoeKer1) [reply src/app/services/assessment.service.ts]: We'll cross that refactor when we get there. Right now I don't see a high chance of that happening.

File: src/app/components/question-progress-bar/question-progress-bar.component.ts (hunk 3 above)
Comment (JoeKer1) [reply src/app/components/question-progress-bar/question-progress-bar.component.ts]: fixed it

**Pull request merged**